src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_Helpers.cs

```diff
@@ -177,30 +160,20 @@ Accessibility.Protected or
     /// Returns <see langword="true"/> if <paramref name="candidate"/> was forwarded to <paramref name="forwardedTo"/> in
     /// <paramref name="forwardedTo"/>'s <see cref="Compilation"/>.
     /// </summary>
-    private static async Task<bool> VerifyForwardedTypeAsync(
+    private static bool VerifyForwardedType(
         Solution solution,
         INamedTypeSymbol candidate,
-        INamedTypeSymbol forwardedTo,
-        HashSet<Compilation> compilationSet,
-        CancellationToken cancellationToken)
+        INamedTypeSymbol forwardedTo)
     {
         // Only need to operate on original definitions.  i.e. List<T> is the type that is forwarded,
         // not List<string>.
         candidate = GetOridinalUnderlyingType(candidate);
         forwardedTo = GetOridinalUnderlyingType(forwardedTo);
 
-        var forwardedToOriginatingProject = solution.GetOriginatingProject(forwardedTo);
-        if (forwardedToOriginatingProject == null)
-            return false;
-
-        var forwardedToCompilation = await forwardedToOriginatingProject.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
+        var forwardedToCompilation = solution.GetOriginatingCompilation(forwardedTo);
```

Comment:
this was the crux of the change down this codepath that made thsi all synchronous.  Previously, teh wrokspae had hte ability to let go of compilations that had been created.  So we had a complex system to try to deal with that, and potentially rehydrate them in the case that that had happened.  but that can't happen here anymore (esp. given that we're asking to get the compilation that produced a particular symbol).